### ISSUE TITLE
Object menu buttons are blank on Safari 15

### ISSUE THREAD

**Issue by takahirox** [mozilla/hubs#4656]
**Description**

Object menu buttons are blank on iOS Safari 15

**To Reproduce**
1. Go to hubs.mozilla.com
2. Enter any room
3. Place any objects
4. Open object menus with double fingers tap
5. Confirm the object menu buttons are blank

**Expected behavior**

The buttons are not blank

**Screenshots**

![image](https://user-images.githubusercontent.com/7637832/134091879-b55382ca-bd52-4b9b-91f6-bc532c70de63.png)

**Hardware** 
 - Device: mobiles
 - OS: iOS 15
 - Browser: Safari 15

**Additional context**

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/HUBS-911)

**Comment by takahirox** [comment]
This problem can be reproducible on my MacOS BigSur 11.6 + Safari 15.

Console errors are

when entering a room

```
WebGL: Error: 0.17:
WebGL: ': extension directive must occur before any non-preprocessor tokens in ESSL3
```

and when opening an object menu

```
WebGL: INVALID_OPERATION: useProgram: program not valid
```

Sounds like Safari15 enabled WebGL2 (and ESSL3) by default but object menu related(?) shader includes some non-ESSL3 compatible code (`extension directive must occur before any non-preprocessor tokens in ESSL3`). Wondering why the problem isn't reproducible on other browsers supporting WebGL2 by default tho.

**Comment by takahirox** [comment]
`aframe/src/shaders/msdf.js` and/or `aframe/src/shaders/sdf.js` seem to cause the problem. It makes sense to me that texts on object menus are not displayed.

I'm investigating how to fix.

**Comment by takahirox** [comment]
**Root issue**

I think I have figured out the root issue.

Our A-Frame's msdf and sdf shaders include

```
'#ifdef GL_OES_standard_derivatives',
'#extension GL_OES_standard_derivatives: enable',
'#endif',
```

https://github.com/MozillaReality/aframe/blob/bb8fe15bca8926d1feb32edf26537594d95ce5a3/src/shaders/msdf.js#L41-L43

And we seemed to have applied [this patch](https://github.com/MozillaReality/aframe/pull/5
) to our A-Frame. The patch seems to insert the shader GLSL3 conversion code to the shaders for WebGL2.

But msdf and sdf shaders include `#extension` and `#extension` must be placed at the top of shader code. So due to the patch, the compilation fails.

If I'm right, [in WebGL2 `GL_OES_standard_derivatives` should be undefined](https://developer.mozilla.org/en-US/docs/Web/API/OES_standard_derivatives). But somehow on Safari it is defined even in WebGL2 (I think it's a bug of Safari). This seems to be the reason why the compilation fails only on the Safari.

**Solutions**

I have two options in my mind.

1. Update msdf and sdf shaders to [the latest ones](https://github.com/aframevr/aframe/blob/master/src/shaders/msdf.js). The latest ones seem to separate WebGL2 shader code and the WebGL2 shader code doesn't include `#extension`. Cons is they use the WebGL2 code anytime if WebGL2 context is available. But `a-scene` takes `webgl2` flag. If webgl2 flag is false, I speculate the compilation can fail. We might be able to check which WebGL version is actually used, but I don't think it's a good design to have dependency with used WebGL context from shader files.

2. Drop the patch. I'm really not sure what the patch is for. If I'm right, GLSL1 is usable even with WebGL2. Locally I dropped the patch and I didn't see any problems so far. And the menu blank object menu problem has been resolved.

I prefer 2. Any thoughts?
